medusa/storage/s3_base_storage.py

```diff
@@ -261,18 +266,49 @@ async def _upload_small_blob(self, src: str, object_key: str):
         await asyncio.sleep(0.001)
 
     async def _upload_big_blob(self, src: str, object_key: str, file_size: int):
+
+        part_size, parts_cnt = self.calculate_part_size(file_size, self.MULTIPART_PART_SIZE, self.MULTIPART_PARTS_COUNT)
+
+        sizes_string = 'file_size={}, part_size={}, parts_cnt={}'.format(
+            _human_readable_size(file_size), _human_readable_size(part_size), parts_cnt
+        )
         logging.debug(
             '[Storage] Uploading multi-part {} ({}) -> s3://{}/{}'.format(
-                src, _human_readable_size(file_size), self.config.bucket_name, object_key
+                src, sizes_string, self.config.bucket_name, object_key
             )
         )
         await self.http_client.put_file_multipart(
             file_path=src,
             object_name=object_key,
             headers=self.additional_upload_headers(),
+            part_size=part_size,
             workers_count=2
         )
 
+    @staticmethod
+    def calculate_part_size(
+            file_size: int,
+            min_part_size: int = DEFAULT_MULTIPART_PART_SIZE,
+            max_parts_cnt: int = DEFAULT_MULTIPART_PARTS_COUNT
+    ) -> t.Tuple[int, int]:
+
+        logging.info(f'Calculating part size for file size {file_size} with max parts count {max_parts_cnt}')
+
+        # check if we fit into the parts limit with the default part size
+        num_parts_with_min_part_size = int(math.ceil(file_size / min_part_size))
+        if num_parts_with_min_part_size < max_parts_cnt:
+            part_size, parts_cnt = min_part_size, num_parts_with_min_part_size
+        else:
+            # if not, calculate the part size that will fit into the parts limit
+            part_size = int(math.ceil(file_size / max_parts_cnt))
+            parts_cnt = max_parts_cnt
```

Comment:
**nit:** IMO it's fine to use `parts_count` instead of `parts_cnt` for better readability of the code. We don't save much with the contraction.